/ros2_laser_scan_merger:
  ros__parameters:
    lidar_topics:
      #- "/mid70/front_scan/scan"
      #- "/mid70/right_scan/scan"
      #- "/mid70/left_scan/scan"
      - "/mid70/front_scan/points"
      - "/mid70/right_scan/points"
      - "/mid70/left_scan/points"
    enabled_lidars:
      - true
      - true
      - true
    target_frame: "mara_top_deck"
    publish_merged_pointcloud: true
    publish_merged_scan: true
    sync: true
    min_range: 0.2
    max_range: 15.0
    voxel_size: 0.05
    angle_min: -1.57 # -0.614356 # 35.2 deg
    angle_max: 1.57 # 0.614356 
    enable_filter_scan: false
    enable_filter_cloud: false  
    
    pointCloudTopic_out: /mid70/merged_points # merged point cloud from all lasers
    laserScanTopic_out: /mid70/merged_scans # merged laserscan from all lasers

    use_sim_time: true

/merged_pointcloud_to_laserscan:
  ros__parameters:
    angle_increment: 0.02
    # angle_max: 1.5708
    # angle_min: -1.5708
    angle_max: 3.14
    angle_min: -3.14
    inf_epsilon: 1.0
    max_height: 10.0
    min_height: 0.0
    qos_overrides:
      /parameter_events:
        publisher:
          depth: 1000
          # durability: volatile
          # history: keep_last
          reliability: best_effort
    queue_size: 10
    range_max: 20.0
    range_min: 0.1
    scan_time: 0.1
    target_frame: mara_top_deck
    transform_tolerance: 0.01
    use_inf: true
    use_sim_time: true
